### PR TITLE
Clientside cppi

### PR DESCRIPTION
**Pull request by thegrb93**

### CODE REVIEW

**Comment by thegrb93** [comment]
Here's my updates. In process of testing

**Comment by thegrb93** [comment]
Seems to be working now.

File: lua/autorun/server/nadmod_pp.lua [hunk 1]
```diff
@@ -382,10 +397,10 @@ function NADMOD.CleanupPlayerProps(steamid)
 	return count
 end
 
-function NADMOD.CleanPlayer(tar)
+function NADMOD.CleanPlayer(ply, tar)
 	if IsValid(tar) and tar:IsPlayer() then 
 		local count = NADMOD.CleanupPlayerProps(tar:SteamID())
-		NADMOD.Notify(tar:Nick().."'s props have been cleaned up ("..count..")")
+		NADMOD.Notify(ply:Nick().. " cleaned up " ..tar:Nick().."'s props ("..count..")")
```
Comment (Nebual) [inline lua/autorun/server/nadmod_pp.lua]: Hmm, `ply` can be null if run from rcon/srcds commands, so we'll want an `ply:IsValid()` check with a fallback "Server" string

**Comment by thegrb93** [comment]
Now it's working

File: lua/autorun/client/cl_nadmodpp.lua [hunk 2]
```diff
@@ -32,7 +68,14 @@ hook.Add("HUDPaint", "NADMOD.HUDPaint", function()
 	if !tr.HitNonWorld then return end
 	local ent = tr.Entity
 	if ent:IsValid() && !ent:IsPlayer() then
-		local text = "Owner: " .. (Props[ent:EntIndex()] or "N/A")
+		local index = ent:EntIndex()
```
Comment (Nebual) [inline lua/autorun/client/cl_nadmodpp.lua]: You'll also want to show 'Owner: Ownerless' + World. 
I also fear this approach means you can't see owner nicknames upon them disconnecting. One solution might be keeping a serverside + clientside mapping of steamID: Last Known Nickname, and falling back to that if they're not ingame anymore.

Also `index` is unused.

File: lua/autorun/client/cl_nadmodpp.lua [hunk 3]
```diff
@@ -23,6 +23,42 @@ net.Receive("nadmod_propowners",function(len)
...
+	if not Props[index] then
+		return false
+	end
+
+	-- Ownerless props can be touched by all
+	if Props[index] == "O" then return true end 
+	-- Admins can touch anyones props + world
+	if NADMOD.PPConfig["adminall"] and NADMOD.IsPPAdmin(ply) then return true end
+	-- Players can touch their own props and friends
+	if Props[index] == ply:SteamID() then return true end
```
Comment (Nebual) [inline lua/autorun/client/cl_nadmodpp.lua]: This isn't checking friends.

Unsure how important false positives `true`'s might be in the Clientside PlayerCanTouch, but the comment at least isn't right anymore.

Theoretically, the client should know about its own friends at least... if we care.

File: lua/autorun/server/nadmod_pp.lua [hunk 4]
```diff
@@ -197,48 +221,11 @@ function NADMOD.GravGunPickup(ply, ent)
...
-		end
-	elseif(mode == "remover") then
-		if(ply:KeyDown(IN_ATTACK2) or ply:KeyDownLast(IN_ATTACK2)) then
-			for k,v in pairs(constraint.GetAllConstrainedEntities(ent) or {}) do
-				if !NADMOD.PlayerCanTouch(ply, v) then
-					return false
-				end
-			end
-		end
-	end
```
Comment (Nebual) [inline lua/autorun/server/nadmod_pp.lua]: Any strong reason why you're removing these edge case handlers + GravGun stuff?

File: lua/autorun/server/nadmod_pp.lua [hunk 5]
```diff
@@ -80,6 +80,41 @@ function NADMOD.IsPPAdmin(ply)
...
+function NADMOD.SendPropOwners(props, ply)
+	net.Start("nadmod_propowners")
+		local seti = 1
+		local set = {}
+		local count = 0
+		for k,v in pairs(props) do
+			if not set[v] then
+				set[v] = seti
+				set[seti] = v
+				seti = seti + 1
```
Comment (Nebual) [inline lua/autorun/server/nadmod_pp.lua]: These variable names are awful haha, but this seems good! I had an earlier comment about optimizing this, and before I posted it, you'd pushed this optimized version :)

**Review by Nebual — CHANGES_REQUESTED** [review]
Nice work on the networking changes!

I can address my comments if you don't want to, though I don't have a dev env setup at the moment

File: lua/autorun/server/nadmod_pp.lua (hunk 4 above)
Comment (thegrb93) [reply lua/autorun/server/nadmod_pp.lua]: I actually meant not to commit those. Damn git.

**Comment by thegrb93** [comment]
I think I resolved them. I'll test again tomorrow.

**Comment by thegrb93** [comment]
Looks like its working.

**Review by Nebual — APPROVED** [review]
Great! Thanks a lot for these improvements! I'll merge this now, and push it to workshop after some testing this weekend

**Comment by thegrb93** [comment]
Thanks

**Pull request merged**